### PR TITLE
Merge into Pranav's PR

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Merge to #16092 when approved.

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-1.0&branch=pr-en-us-16129&tabs=visual-studio#authorization-for-specific-endpoints)

External review:

* [Download 1.zip](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
* Unzip 1.zip file

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@serpent5  please review.

**Comment by serpent5** [comment]
As a more general comment, there's some really useful information here re authz that's not specific to the migration path. I wouldn't expect to come to this particular page to learn out about things like `DefaultPolicy`, `FallbackPolicy`, etc.

I wonder if it would be good to move some of this information out to the authz pages under "Security and Identity" and just cover the migration from the global filter here.

**Comment by Rick-Anderson** [comment]
> As a more general comment, there's some really useful information here re authz that's not specific to the migration path. I wouldn't expect to come to this particular page to learn out about things like `DefaultPolicy`, `FallbackPolicy`, etc.

> I wonder if it would be good to move some of this information out to the authz pages under "Security and Identity" and just cover the migration from the global filter here. Clearly, if that's deemed to be a good idea, it wouldn't be part of this work.
That's a great idea. Do you want to open an issue for that. We can merge this in the mean time then move it over.

File: aspnetcore/migration/22-to-30/samples/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                     options => options.SignIn.RequireConfirmedAccount = true)
+                .AddEntityFrameworkStores<ApplicationDbContext>();
+
+            // Pre 3.0:
+            // services.AddMvc(options => options.Filters.Add(new AuthorizeFilter(...));
+
+            services.AddControllersWithViews();
+            services.AddRazorPages();
+            services.AddAuthorization(o => o.AddPolicy(_RequireAuthenticatedUserPolicy,
+                            builder => builder.RequireAuthenticatedUser()));
```
[the remark this answers is not in the document]
Comment (pranavkm) [reply aspnetcore/migration/22-to-30/samples/Startup.cs]: I wanted to show a use of `RequireAuthorization`. `RequireAuthenticatedUser` was easy to set up, but feel free to add some more to this if you think that helps

File: aspnetcore/migration/22-to-30.md [hunk 2]
```diff
@@ -688,7 +688,7 @@ public void Configure(IApplicationBuilder app)
     });
 }
 
-public class HomeController : ControllerBase
+public class HomeController : Controller
```
Comment (pranavkm) [inline aspnetcore/migration/22-to-30.md]: Why change this?

Comment (Rick-Anderson) [reply aspnetcore/migration/22-to-30.md]: > Why change `ControllerBase` to `Controller`

That's what the templates generate so I assume that's what most customers use.

**Pull request merged**